docs/modules/candid-guide/index.adoc

```diff
@@ -0,0 +1,1098 @@
+  add : (nat) -> ();
+  subtract : (nat) -> ();
+  get : () -> (int) query;
+  subscribe : (func (int) -> ()) -> ();
+}
+....
+This can evolve to the following interface:
+....
+type timestamp = nat;
+service counter : {
+  set : (nat) -> ();
+  add : (int) -> (new_val : nat);
+  subtract : (nat, trap_on_underflow : opt bool) -> (new_val : nat);
+  get : () -> (nat, last_change : timestamp) query;
+  subscribe : (func (nat) -> (unregister : opt bool)) -> ();
+}
+....
+====
+
+[#textual-values]
+=== Candid textual values
+
+The main purpose of Candid is to connect programs written in some host language (Motoko, Rust, JavaScript, etc.), and developers therefore usually do not have to deal with “Candid values” per se: They work with, say, JavaScript values that are then, transparently and seamelessly, transported to a canister written in Rust, where they show up as Rust values.
+
+Nevertheless there are use cases where it is helpful to see the Candid values per se, e.g. for logging, debugging, or command-line interaction with services. In this cases, the _textual presentation_ for Candid values can be used.
+
+The syntax is similar to that of candid types. A typical candid value might read
+....
+(record {
+  first_name = "John";
+  last_name = "Doe";
+  age = 14;
+  membership_status = variant { active };
+  email_addresses =
+    vec { "john@doe.com"; "john.doe@example.com" };
+})
+....
+
+[NOTE]
+====
+The Candid _binary_ format does not include the actual field names, merely numeric _hashes_. So pretty-printing such a value without knowledge of the expected type will not include the field names of records and variants. The above value might then be printed as
+....
+(record {4846783 = 14; 456245371 = variant {373703110}; 1443915007 = vec {"john@doe.com"; "john.doe@example.com"}; 2797692922 = "John"; 3046132756 = "Doe"})
+....
+====
+
+=== Generating service descriptions
+
+In the <<candid-service-descriptions,section above>> you learned how to write a Candid service description from scratch. But often, that is not even needed! Depending on the language you use to implement your service, you can get the Candid service description generated from your code.
+
+For example, in Motoko, you can write a canister as
+....
+actor {
+  var v : Int = 0;
+  public func add(d : Nat) : async () { v += d; };
+  public func subtract(d : Nat) : async () { v -= d; };
+  public query func get() : async Int { v };
+  public func subscribe(handler : func (Int) -> async ()) { … }
+}
+....
+and it will have the interface shown above.
+
+In other languages, like Rust, you can still develop your service while only
+touching Rust types, but for now you still have to write the Candid interface description manually.
+
+In either case it is helpful to know the mapping between host language types and Candid types; this mapping is describeed in the <<reference, reference section>>.
+
+=== Candid tools and libraries
+
+The Motoko language has built-in support for Candid.
```

Comment:
```suggestion
The Motoko language has built-in support for Candid. A Motoko program can import a service as a Motoko actor, just by consuming the Candid of that service. A Motoko actor can implement some Candid service by exporting a Candid description of its Motoko interface.
```

Reply:
I’d rather add that, with actual instructions, to “How do I use Candid as a Motoko canister developer?”. Although unclear where to draw the line between the candid manual and the Motoko manual.